### PR TITLE
Fix KW scan issues

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: sound/soc/sof/topology.c [hunk 1]
```diff
@@ -2506,6 +2506,9 @@ void snd_sof_free_topology(struct snd_sof_dev *sdev)
 		sof_route_remove(&sroute->route);
 
 		/* free sroute and its private data */
+		kfree(sroute->route.source);
+		kfree(sroute->route.sink);
+		kfree(sroute->route.control);
```
Comment (plbossart) [inline sound/soc/sof/topology.c]: I am not quite happy with this free_topology function. Just like we have a mirror between snd_soc_tplg_component_load and snd_soc_tplg_component_remove, it'd be easier if we could identify where all the sof routes are allocated. I don't know how one matches what is freed with what was allocated?

Also this code does a kfree(sroute->route.source) after sof_route_remove frees route->source so something looks very odd here.

Comment (ranj063) [reply sound/soc/sof/topology.c]: @plbossart I agree. I am still not finished looking into the topology_free() in detail to make the call here. Maybe the kwscan issue will have to wait until that it cleaned up first then.

**Comment by ranj063** [comment]
@plbossart i've removed the sroute freeing part. I'll update it later.

**Comment by ranj063** [comment]
@plbossart can we please merge this patch. This will help with the glk release. Thanks!

**Pull request merged**